packages/taquito/src/prepare/prepare-provider.ts

```diff
@@ -1098,4 +1101,78 @@ export class PrepareProvider extends Provider implements PreparationProvider {
       contents,
     };
   }
+
+  /**
+   *
+   * @description Method to prepare a stake operation
+   * @param operation RPCOperation object
+   * @param source string or undefined source pkh
+   * @returns a PreparedOperation object
+   */
+  stake(operation: StakingParams, source?: string): Promise<PreparedOperation> {
```

Comment:
previous comments that would be great that source to be optional in StakingParams. 
it's fine if you don't like to follow the rest of ops defined parameter {fee, storageLImit, gasLImit, ...rest}, think the `operation: StakingParams` of operation can be better named.